src/main/java/microservices/userServiceBasic/api/UserApiRest.java

```diff
@@ -0,0 +1,31 @@
+package microservices.userServiceBasic.api;
+
+import microservices.userServiceBasic.api.mapper.UserApiMapper;
+import microservices.userServiceBasic.api.model.UserDetails;
+import microservices.userServiceBasic.api.response.UserResponse;
+import microservices.userServiceBasic.domain.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class UserApiRest implements UserApi {
+
+	private UserService userService;
+
+	@Autowired
+	public UserApiRest(UserService userService) {
+		this.userService = userService;
+	}
+
+	@Override
+	@RequestMapping(value = "/user", method = RequestMethod.POST, produces = "application/json")
+	public ResponseEntity<UserResponse> createUser(@RequestBody UserDetails user) {
+
+		userService.saveUser(UserApiMapper.mapToUser(user));
```

Comment:
`UserApiMapper` should be injected into this class and it Should not be called statically.

Reply:
`userService.saveUser(...)` should return the saved user that should be mapped to the UserResponse to be outputted.